### PR TITLE
Bump CakePHP version to 4.2

### PR DESCRIPTION
**Pull request by ishanvyas22**
### Fixed

- Failing unit tests

### Improved

- Bump minimum CakePHP version to 4.2
- Support for PHP 8.2
- Replaced travis CI with Github Actions

### CODE REVIEW

File: config/schema/articles.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+CREATE TABLE `bryancrowe_articles` (
```
Comment (pabloelcolombiano) [inline config/schema/articles.sql]: Maybe something to place under the tests domain?

Comment (ishanvyas22) [reply config/schema/articles.sql]: @pabloelcolombiano thanks for the review. I've moved the file.

**Comment by ishanvyas22** [comment]
@bcrowe Can you please have a look at this?

File: config/schema/articles.sql (hunk 1 above)
Comment (pabloelcolombiano) [reply config/schema/articles.sql]: Thanks, looks good to me!

**Comment by bcrowe** [comment]
Hey, sorry for the delay! Let me take a look over this -- also I need to get the builds running again, not sure what happened there. Sit tight!

Edit: Oh, I see you took care of the CI build... thank you!

**Comment by bcrowe** [comment]
I think my only question is, should we instead use `^4.2` & `>=7.2` for the dependencies? Looks like PHP 8+ support started with 4.2. @ishanvyas22  @pabloelcolombiano 

From what I see here: https://github.com/cakephp/cakephp/wiki#version-map

**Comment by bcrowe** [comment]
~~One more thing. Maybe I'm missing something -- but wouldn't this continue to work fine on PHP 8+ and Cake 4.4, without needing to bump up the plugin's dependencies' versions? I don't think I see anything being introduced that isn't backwards compatible with 4.0 through 4.3. I guess I'm not seeing a need to release a new version of this that requires 4.4.~~

Although, it's definitely great that we have a working CI build again and it's being tested against PHP 8 now as well. And taking care of the deprecated fixture stuff.

Let me know if I've missed something.

Edit: Oh, I guess for running the tests against PHP 8+ it would have to be a version of Cake that supports it. 😅 Okay, just curious about my question about `^4.2` & `>=7.2`  then. Also, feel free to delete the TravisCI config file too.

**Comment by bcrowe** [comment]
@ishanvyas22 @pabloelcolombiano I'll be checking this often so it doesn't get any further delayed. Let me know what you think.

**Comment by bcrowe** [comment]
And let me know if you have Discord -- throwing me a message there anytime you make a PR in the future would get my eyes on things right away. Tons of notifications/emails and sometimes these slip by me for a little bit.

**Comment by ishanvyas22** [comment]
Hey @bcrowe, thanks for taking a look. I've changed min CakePHP requirements to `^4.2` and PHP at `^7.2`.

**Comment by bcrowe** [comment]
LGTM! Thanks for the edits and getting the CI build up and running. I'll release/tag this as a new major version `3.0.0`.

**Comment by bcrowe** [comment]
https://github.com/bcrowe/cakephp-api-pagination/releases/tag/3.0.0

**Comment by ishanvyas22** [comment]
Thanks @bcrowe 🙏🏼

**Pull request merged**